client/components/common/Header.js

```diff
@@ -1,53 +1,23 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { Link } from 'react-router';
 
-import { signOut } from "../../actions/userActions";
+import NavBar from './NavBar';
 
-class Header extends Component {
-  constructor(props){
-    super(props);
-    this.onSignOut = this.onSignOut.bind(this);
-  }
+import { signOut } from '../../actions/userActions';
 
-  onSignOut(){
+class Header extends Component {
+  onSignOut = () => {
     this.props.signOut();
     window.location.reload();
```

Comment:
### Nitpick

I believe we can do better than reloading the browser, let's redirect the user using the router `Redirect` component or push location directly.